docs/guides/testing/testing.rst

```diff
@@ -117,143 +117,67 @@ Note -
 
     paver -h
 
+Note -
+Unless otherwise mentioned, all the following commands should be run from inside lms docker container.
 
 Running Python Unit tests
 -------------------------
 
-We use `pytest`_ to run the test suite.
+We use `pytest`_ to run python tests. Pytest is a testing framework for python and should be your goto for local python unit testing.
 
 .. _pytest: https://pytest.org/
 
-For example, this command runs all the python test scripts::
-
-    paver test_python
-
-It also runs ``collectstatic``, which prepares the
-static files used by the site (for example, compiling Sass to
-CSS).
-
-You can re-run all failed python tests by running this command (see note at end of
-section)::
-
-    paver test_python --failed
-
-To test lms python tests use this command::
-
-    paver test_system -s lms
-
-To test cms python tests use this command::
-
-    paver test_system -s cms
-
-To run these tests without ``collectstatic``, which is faster, append the following argument::
-
-    paver test_system -s lms --fasttest
-
-To run cms python tests without ``collectstatic`` use this command::
-
-    paver test_system -s cms --fasttest
-
-For the sake of speed, by default the python unit test database tables
-are created directly from apps' models. If you want to run the tests
-against a database created by applying the migrations instead, use the
-``--enable-migrations`` option::
-
-    paver test_system -s lms --enable-migrations
-
-To see the migration output, use::
-
-    paver test_system -s lms --enable-migrations --verbose --disable_capture
-
-To run a single django test class use this command::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest
 
 Running a Single Test
 ~~~~~~~~~~~~~~~~~~~~~
 
-When developing tests, it is often helpful to be able to really just run
-one single test without the overhead of PIP installs, UX builds, etc. In
-this case, it is helpful to look at the output of paver, and run just
-the specific command (optionally, stripping away coverage metrics). At
-the time of this writing, the command is the following::
-
-    pytest lms/djangoapps/courseware/tests/test_courses.py
-
-
-To run a single test format the command like this::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest::test_activate_login
-
-You can use ``--randomize`` to randomize the test case sequence.  In the
-short term, this is likely to reveal bugs in our test setup and teardown;
-please fix (or at least file tickets for) any such issues you encounter.
-
-You can also enable test concurrency with the ``--processes=N`` flag (where ``N``
-is the number of processes to run tests with, and ``-1`` means one process per
-available core). Note, however, that when running concurrently, breakpoints may
-not work correctly.
```

Comment:
`pytest-xdist` lets you do this with `-n` or `--numprocesses`.  It's useful to use `-n=0` sometimes when you want to force all tests to run in a single process.  Here's the whole help section from `pytest --help`:
```
  -n numprocesses, --numprocesses=numprocesses
                        shortcut for '--dist=load --tx=NUM*popen', you can use 'auto' here for auto detection CPUs number on host
                        system and it will be 0 when used with --pdb
```

Reply:
I do not know the best way to include this info in this PR, so I ask you to add this point in a later PR.